src/services/ProcessWalletEvent.js

```diff
@@ -0,0 +1,106 @@
+import { WALLET_EVENTS } from '@/constants'
+
+export default class NetworkStatus {
+  constructor() {
+    this.currentState = null
+    this.isResyncButtonVisible = false
```

Comment:
This service shouldn't know anything about UI, but I think we can create a method such as `isResyncAvailabe` where we calculate it.